api/src/main/java/org/pmiops/workbench/api/CohortReviewController.java

```diff
@@ -133,6 +137,15 @@
                                                                                       Long cohortId,
                                                                                       Long cdrVersionId,
                                                                                       CreateReviewRequest request) {
+        // TODO: enforce access level.
+        // This also enforces registered auth domain.
+        WorkspaceAccessLevel accessLevel;
+        try {
+          accessLevel = workspaceService.getWorkspaceAccessLevel(workspaceNamespace, workspaceId);
+        } catch (Exception e) {
+          throw e;
+        }
```

Comment:
A better place for this code may be in the CohortReviewServiceImpl#validateMatchingWorkspace. Were already validating that workspace in this method and it can be called from all the methods needed in the CohortReviewController. https://github.com/all-of-us/workbench/blob/blrubenstein/enforce-auth-domain/api/src/main/java/org/pmiops/workbench/cohortreview/CohortReviewServiceImpl.java#L71-L78

Reply:
Also, we could potentially add the CohortBuilderController#countParticipants, CohortBuilderController#countParticipants#getChartInfo and CohortBuilderController#getCriteriaTreeQuickSearch. Not sure if we need to add these but I would assume that even if you could hit these without access.. you won't be able to save a cohort anyway.

Reply:
Also, we could explore if an interceptor is a better way of checking this rather than checking access level in every api call within the workbench domain.